pyphare/pyphare/pharesee/plotting.py

```diff
@@ -4,6 +4,100 @@
     BboxPatch, BboxConnector, BboxConnectorPatch)
 
 
+import numpy as np
+import matplotlib.pyplot as plt
+
+def dist_plot(particles, **kwargs):
+    """
+    plot the phase space of given particles
+    particles can be of type Particles, list(Particles), dict{popname:Particles}
+
+    kwargs:
+    * axis : ("x", "Vx"), ("x", "Vy"), ("x", "Vz"), ("Vx", "Vy") (default) --
+       ("Vx", "Vz"), ("Vy", "Vz")
+    * bins :  number of bins in each dimension, default is (50,50)
+    * sigma : width of the gaussian filter, default is (0,0)
+    * norm  : histogram will be normed to Normalize(0,norm)
+    * kde : (default False) : adds contours of kernel density estimate
+    * title : (str) title of the plot
+    * xlabel, ylabel
+    * xlim, ylim
+    * bulk : (bool) (default : False), adds vertical/horizontal lines --
+             at in-plane bulk velocity for velocity axis
+    * filename : (str) if exists, save plot to figure under that name
+
+    return value : fig,ax
+    """
+    from pyphare.pharesee.particles import Particles, aggregate
+    from matplotlib.colors import Normalize
+
+    if isinstance(particles, list):
+        particles = aggregate(particles)
+    elif isinstance(particles, dict):
+        particles = aggregate([p for p in particles.values()])
+
+    if not isinstance(particles, Particles):
+        raise ValueError("Error, 'particles' type should be Particles, list or dict")
+
+    from scipy.ndimage import gaussian_filter as gf
+    if "ax" not in kwargs:
+        fig, ax = plt.subplots()
+    else:
+        ax = kwargs["ax"]
+        fig = ax.figure
+    axis = kwargs.get("axis",("Vx","Vy"))
+    vaxis = {"Vx":0, "Vy":1, "Vz":2}
+
+    if axis[0] in vaxis:
+        x = particles.v[:,vaxis[axis[0]]]
+    elif axis[0] == "x":
+        x = particles.x
+    if axis[1] in vaxis:
+        y = particles.v[:,vaxis[axis[1]]]
+
+    bins = kwargs.get("bins", (50,50))
+    h, xh, yh  = np.histogram2d(x, y,
+              bins=kwargs.get("bins", bins),
+              weights=particles.weights/particles.dl[:,0])
```

Comment:
I agree with this, but it looks weird : `particles.weights` looks like a weight and `weights` like a density... with somewhat the same name.

Reply:
Not sure I understand your comment. I've removed the division by the dl anyway since this PR is now based on #466